include/envoy/http/codec.h

```diff
@@ -260,6 +283,16 @@ class ClientConnection : public virtual Connection {
    * @return StreamEncoder& supplies the encoder to write the request into.
    */
   virtual StreamEncoder& newStream(StreamDecoder& response_decoder) PURE;
+
+  /**
+   * Called when the connection goes over its high watermark.
```

Comment:
Fine with a longer name.  would onUnderlyingConnectionAboveWriteBufferHighWatermark() or onUnderlyingNetworkConnectionAboveWriteBufferHighWatermark() work as well for you?

If "backing" is more envoyish I'll go with that.  I'm still adapting to project nomenclature :-)